### ISSUE TITLE
Pass extension arguments to refactor invocation

### ISSUE THREAD

**Issue by pbrisbin** [ndmitchell/hlint#1118]
I'm a heavy user of HLint's apply-refact auto-fixing through my [Restyled](https://restyled.io) project, as well as editor integrations (in my case, vim's ALE plugin).

I frequently run into errors such as:

```
Restyling "src/Restylers/Check.hs" via "hlint"
refactor: (RealSrcSpan SrcSpanOneLine "./src/Restylers/Check.hs" 23 16 20,"Illegal lambda-case (use -XLambdaCase)")
CallStack (from HasCallStack):
  error, called at src/Refact/Run.hs:190:22 in main:Refact.Run
```

This is because I've used a syntactic extension set via `default-extensions`. HLint itself is fine with this as long as I mirror that in _its_ configuration file, which I do:

```yaml
- arguments:
  # ...
  - -XLambdaCase
```

But that fact never makes it to `apply-refact`, nor is there any equivalent configuration file (that I can find). To make this work, I have to specify that particular extension via `LANGUAGE`.

This is not ideal. We like to use a uniform set of default-extensions across all projects at `$work` (as is common) and we consider this one default. I can't choose to set it via `LANGUAGE`, which impacts everyone else interacting with the source files, just to satisfy my own tooling choices.

apply-refact appears to accept [extension arguments](https://hackage.haskell.org/package/apply-refact):

> Multiple -X options may be provided to specify additional default language pragmas which might affect parsing, such as -XLambdaCase or -XRankNTypes.

So I wonder (assuming I'm not missing an existing way to accomplish this), can HLint take its own `arguments` configuration, filter to any arguments matching `/^-X/`, and pass them along when invoking `refactor`?

It's already a hassle to keep `package.yaml` and `hlint.yaml` in sync (along with `brittany.yaml` and `stylish-haskell.yaml` :dizzy_face: ) with respect to extensions. It would be *wonderful* if at least hlint-to-apply-refact Just Worked.

**Comment by ndmitchell** [comment]
#1027 aimed to do roughly that, which was released as v 3.1.5 on 2020-06-19. Is your HLint newer than that?

**Comment by pbrisbin** [comment]
Oh jeeze, I've not looked at Restyled's version in a while and it's on 2.1.11. I'll trust it works and re-open if I find otherwise, sorry for the noise! (I did search, but I must've done something incorrectly there.)
